example/fsm_example/models.py

```diff
@@ -1,37 +1,55 @@
+from pprint import pprint
+
 from django.db import models
+from django.dispatch.dispatcher import receiver
 from django.utils import timezone
+from django_fsm import FSMField, transition, GET_STATE, RETURN_VALUE, ConcurrentTransitionMixin
+from django_fsm.signals import pre_transition
 
-from django_fsm import FSMField, transition
-
-
-class State(object):
-    '''
-    Constants to represent the `state`s of the PublishableModel
-    '''
-    DRAFT = 'draft'            # Early stages of content editing
-    APPROVED = 'approved'      # Ready to be published
-    PUBLISHED = 'published'    # Visible on the website
-    EXPIRED = 'expired'        # Period for which the model is set to display has passed
-    DELETED = 'deleted'        # Soft delete state
-
-    CHOICES = (
-        (DRAFT, DRAFT),
-        (APPROVED, APPROVED),
-        (PUBLISHED, PUBLISHED),
-        (EXPIRED, EXPIRED),
-        (DELETED, DELETED),
-    )
+from .utils import ModelEnum
+
+
+class State(ModelEnum):
+    """Constants to represent the `state`s of the PublishableModel"""
+    DRAFT = 'DRAFT'            # Early stages of content editing
```

Comment:
Django-fsm-admin does not support different name and values. Or is it CharField?